python/lsst/ts/watcher/mock_pager_duty.py

```diff
@@ -0,0 +1,239 @@
+from __future__ import annotations
+
+# This file is part of ts_watcher.
+#
+# Developed for Vera C. Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+__all__ = ["MockPagerDuty"]
+
+import os
+import types
+import typing
+import uuid
+from http import HTTPStatus
+
+from aiohttp import web
+
+
+def get_event_key() -> str:
+    """Return a new unique request key"""
+    return str(uuid.uuid4())
+
+
+class MockPagerDuty:
+    """A mock PagerDuty service to support Watcher escalation in unit tests.
+
+    Parameters
+    ----------
+    port : `int`
+        The TCP/IP port for the service. 0 to choose a free port.
+
+    Notes
+    -----
+    To use::
+
+        async with MockPagerDuty(port=...) as server:
+            # ... use the server
+
+        # Or if you prefer explicit start/stop:
+
+        server = MockPagerDuty(port=...)
+        await server.start()
+        # ... use the server
+        await server.close()
+
+    Attributes
+    ----------
+    port : `int`
+        If the specified port is 0, then the chosen port
+        (after `start` has run). Otherwise the specified port.
+        The default is 80 because that is what PagerDuty uses.
+    url : `str`
+        The root URL of the service. "" until `start` has run.
+    """
+
+    ALLOWED_KEYS = frozenset(
+        (
+            "payload",
+            "routing_key",
+            "event_action",
+            "dedup_key",
+            "client",
+            "client_url",
+            "links",
+            "images",
+        )
+    )
+
+    ALLOWED_PAYLOAD_KEYS = frozenset(
+        (
+            "summary",
+            "timestamp",
+            "severity",
+            "source",
+            "component",
+            "group",
+            "class",
+            "custom_details",
+        )
+    )
+
+    def __init__(self, port: int = 80) -> None:
+        self.port = port
+        self.escalation_key = os.environ["ESCALATION_KEY"]
+        # Dict of event ID: data_dict
+        self.events: dict[str, dict] = dict()
```

Comment:
Can you be more explicit about `data_dict`? What does that mean? Any example or a URL that points to one?